src/components/tl/ucp/tl_ucp_lib.c

```diff
@@ -27,9 +27,19 @@ UCC_CLASS_CLEANUP_FUNC(ucc_tl_ucp_lib_t)
 
 UCC_CLASS_DEFINE(ucc_tl_ucp_lib_t, ucc_tl_lib_t);
 
-ucc_status_t ucc_tl_ucp_get_lib_attr(const ucc_base_lib_t *lib, /* NOLINT */
-                                     ucc_base_attr_t *attr)     /* NOLINT */
+ucc_status_t ucc_tl_ucp_get_lib_attr(const ucc_base_lib_t *lib,
+                                     ucc_base_attr_t *base_attr)
 {
-    //TODO
-    return UCC_ERR_NOT_IMPLEMENTED;
+    ucc_tl_lib_attr_t *attr = ucc_derived_of(base_attr, ucc_tl_lib_attr_t);
+    ucs_status_t status;
+    ucp_lib_attr_t params;
+    memset(&params, 0, sizeof(ucp_lib_attr_t));
+    params.field_mask = UCP_LIB_ATTR_FIELD_MAX_THREAD_LEVEL;
+    status            = ucp_lib_query(&params);
+    if (status != UCS_OK) {
+        ucc_error("failed to query UCP lib attributes");
+        return UCC_ERR_NO_RESOURCE;
+    }
+    attr->super.attr.thread_mode = (ucc_thread_mode_t) params.max_thread_level;
```

Comment:
you can't cast this way. ucc_threawd_mode may be different from ucc_thread_mode_t